Docs website: Added CD, added node script,  some hirarchy changes

liorsve opened the pull request:
<!--
Thanks for contributing to Valkey GLIDE!

Please make sure you are aware of our contributing guidelines [available
here](https://github.com/valkey-io/valkey-glide/blob/main/CONTRIBUTING.md)

-->

### Issue link

This Pull Request is linked to issue : #2377, #2725

This PR - 
1. adds a workflow to build and deploy the website to the valkey-glide github pages.
2. adds a script to the `node/build-docs` script that reformats all of the links in the node docs.
3. adds some changes to website layout - 
   - removes the node "commands" section from the main nav- I thought it's a but confusing with the commands not really being there. The links these classes from other pages don't break. 
   - adds `command`, `config`, `logger` sections to python docs, adds `config` section to node  (in an attempt to make the two navs look more similar to one another.

Preview from my gh pages - https://liorsve.github.io/valkey-glide/


### Checklist

Before submitting the PR make sure the following are checked:

-   [x] This Pull Request is related to one issue.
-   [x] Commit message has a detailed description of what changed and why.
-   [x] Tests are added or updated.
-   [x] CHANGELOG.md and documentation files are updated.
-   [x] Destination branch is correct - main or release
-   [x] Create merge commit if merging release branch into main, squash otherwise.

Yury-Fridlyand reviewed (dismissed):
python client docs are empty

edlng commented:
Ran the docs on my local machine and it looks quite nice! I have a few comments/requests however:

1. When using the side bar to navigate through commands, there's this page in between that's a little annoying. Can this be removed?

After clicking on one of the commands, i have to click on `index.html` to get to the actual page
![image](https://github.com/user-attachments/assets/f36cceb0-4764-4989-ac72-972341d1b817)
![image](https://github.com/user-attachments/assets/7948d815-6b9d-451b-9690-d9a55b7b1877)
![image](https://github.com/user-attachments/assets/1015dd76-074e-42e4-a3b2-a4908df699e9)

2. We should add unnecessary files/folders to `.gitignore` after building (about 424 files were added on my local machine).

3. nit: but as seen in the picture above, methods are named "meth". Is there a way to configure this?

edlng commented:
Thanks @liorsve !
1. Ah okay if it doesn't happen in your deployment then it should be fine. Probably some configuration I have to tweak locally.
2. Locally, it's creating this path `docs/site` and inside it created all the `index.html` files and asset files and etc. I noticed in the `.gitignore` we don't ignore this path so maybe we need to add it.
3. No worries!
4. Sounds good :)

Also as a side note, when I run `./build-docs.sh build`, it modifies `node/package.json` to change this `"typedoc": "^0.28.x" ` to `"typedoc": "^0.28.2"`. Can we prevent this change? If not maybe we should fully change it to `"^0.28.2"` instead?

Yury-Fridlyand commented on node/build-docs:
```diff
@@ -15,3 +14,43 @@ typedoc src/BaseClient.ts                   \
+Create a temporary Node.js script for link conversion
+This is a workaround to convert the links with the @link syntax to markdown links
+cat > convert-links.mjs << 'EOL'
+import * as fs from 'fs';
```
keep this as a file

Yury-Fridlyand commented on .github/workflows/mkdocs-deploy.yml:
```diff
@@ -0,0 +1,46 @@
+                    mkdocstrings-python==1.13.0 \
+                    pymdown-extensions \
+                    mkdocs-breadcrumbs-plugin \
+                    mkdocs-material \
```
```suggestion
                    mkdocs-material
```

Yury-Fridlyand commented on .github/workflows/mkdocs-deploy.yml:
```diff
@@ -0,0 +1,46 @@
+                    mkdocs-breadcrumbs-plugin \
+                    mkdocs-material \
+
+            - name: Make script executable
+              run: chmod +x ./docs/build-docs.sh
+
```
```suggestion
```
commit this script with `+x` permissions instead

eifrah-aws replied:
It already has execution permissions in git, do we need this `chmod +x`?

liorsve replied:
removed

Yury-Fridlyand commented on docs/README.md:
```diff
@@ -0,0 +1,23 @@
+
+- **Python documentation** is generated using the [`mkdocstrings`](https://mkdocstrings.github.io/) plugin for MkDocs.
+- **Node.js documentation** is generated using [`TypeDoc`](https://typedoc.org/), which produces Markdown files that are then processed by MkDocs.
+
```
Add link to python's dev guide to a section related to documentation.

eifrah-aws commented on node/build-docs:
```diff
@@ -15,3 +14,43 @@ typedoc src/BaseClient.ts                   \
         --disableSources                    \
         --includeHierarchySummary           \
         --out ../docs/markdown/node/
+Create a temporary Node.js script for link conversion
```
Is this line (and the one below it) are comments? if so, where is the `#`?

liorsve replied:
removed

eifrah-aws commented on node/build-docs:
```diff
@@ -1,6 +1,5 @@
 #!/bin/bash
```
Please add here `-e`

liorsve replied:
added

liorsve commented:
- Addressed comments 
- Made a separate `node/docs` directory for the 3 files related to typedoc.
- Added the seperate `convert-urls.mjs` script, had to make some changed to the script to avoid security issued, and yet had to introduce some `//nosemgrep` comments.

eifrah-aws commented on node/docs/build-docs:
```diff
@@ -0,0 +1,20 @@
+
+
+# Run urls conversion script - makes them apear not broken in the website
+node ./docs/convert-urls.mjs
```
do we have node installed on the machine or do we need to make sure to install it in the CI?
